pkg/validate/serialize.go

```diff
@@ -17,28 +33,181 @@ func Verify(yamlFileName string) error {
 	}
 
 	// Value returned is a marshaled go type (CSV Struct).
-	_, err = unmarshal(rawYaml)
+	csv, err := unmarshal(rawYaml)
 	if err != nil {
 		return fmt.Errorf("Error unmarshalling YAML to OLM's csv type for %s file:  #%s ", yamlFileName, err)
 	}
 
-	fmt.Printf("%s is verified", yamlFileName)
+	// Contains error logs for all missing optional and mandatory fields.
+	errorLog := csvInspect(csv)
+	getErrorsFromManifestResult(errorLog.warnings)
+
+	// There is no mandatory field thats missing if errorLog.errors is nil.
+	if errorLog.errors != nil {
+		getErrorsFromManifestResult(errorLog.errors)
+		return fmt.Errorf("Populate all the mandatory fields missing from %s file.", yamlFileName)
+	}
+	fmt.Printf("%s is verified.", yamlFileName)
 	return nil
 }
 
+// Iterates over the list of warnings and errors.
+func getErrorsFromManifestResult(err []missingTypeError) {
+	if err != nil {
+		for _, v := range err {
```

Comment:
what happens if err is nil here?

Reply:
No error/warning gets printed. Control goes back and you just see `csv.yaml file is verified.`

Reply:
Do I need to explicitly `return` at the end if err is nil?